disperser/controller/dispatcher.go

```diff
@@ -304,16 +309,20 @@ func (d *Dispatcher) HandleBatch(ctx context.Context) (chan core.SigningMessage,
 	return sigChan, batchData, nil
 }
 
-// HandleSignatures receives signatures from operators, validates, and aggregates them
+// HandleSignatures receives signatures from operators, validates, and aggregates them.
+//
+// This method submits Attestations to the blobMetadataStore, containing signing data from the SigningMessages received
+// through the sigChan. It periodically submits Attestations, as signatures are gathered.
 func (d *Dispatcher) HandleSignatures(
```

Comment:
whats the termination criterion for this function? does it just loop forever? until some timer ends? is it run in a goroutine?

Reply:
This function doesn't loop, it's called once per batch. I reworked this doc in f3b5f28bf17e140835357144ea35be82e44d7e28 to improve clarity

```
// HandleSignatures receives SigningMessages from operators for a given batch through the input sigChan. The signatures
// are validated, aggregated, and used to put an Attestation for the batch into the blobMetadataStore. The Attestation
// is periodically updated as additional signatures are gathered.
//
// This method will continue gathering signatures until a SigningMessage has been received from every operator, or until
// the global attestationCtx times out.
```